packages/x-ui-admin/src/react/components/RemoteSelect.tsx

```diff
@@ -1,17 +1,45 @@
 import * as React from "react";
 import { Constructor } from "@bluelibs/core";
-import { Collection, use, useData } from "@bluelibs/x-ui";
+import { Collection, QueryBodyType, use, useData } from "@bluelibs/x-ui";
 import { Alert, Select, SelectProps, Spin } from "antd";
 import { ObjectId } from "@bluelibs/ejson";
 
+export interface OnSelectDataChangeOptions {
```

Comment:
I remember we also discussed about having a `value` resolver based on the data fetched. I think we need to explore the following route:

```tsx
<RemoteSelect body={{ name: 1, email: 1, }} documentToLabel={(doc) => doc.name} 
```

Reply:
Shall I fully remove the `field` prop then ?

Reply:
I think the solution is this, we either specify `field` as a string, and it gets into the body request, and if there's no `documentToLabel={}` we render the label document[field].

We have to do a check. If there isn't a field, documentToLabel is required so we know how to render the thing.

Reply:
@Fen747 any input? I think it's almost done.